src/domain.cc

```diff
@@ -793,6 +813,117 @@ NLV_WORKER_EXECUTE(Domain, ToXml)
   free(result);
 }
 
+NAN_METHOD(Domain::GetMetadata)
+{
+  NanScope();
+  int type;
+  std::string namespace_uri;
+  unsigned int flags;
+  NanCallback *callback;
+#ifndef _HAVE_DOMAIN_METADATA_API
+  NanThrowTypeError("metadata api not supported in this libvirt version");
+  NanReturnUndefined();
+#endif
+  if (args.Length() == 4
+	  && args[0]->IsNumber()
+	  && (args[1]->IsString() || args[1]->IsNull())
+	  && args[2]->IsNumber()
+	  && args[3]->IsFunction())
+  {
+    type = args[0]->IntegerValue();
+    if (!args[1]->IsNull()) {
+	namespace_uri = *NanUtf8String(args[1]->ToString());
+    }
+    flags = args[2]->IntegerValue();
+    callback = new NanCallback(args[3].As<Function>());
+  } else {
+    NanThrowTypeError("signature is type, namespace_uri, flags, callback");
+    NanReturnUndefined();
+  }
+  Domain *domain = ObjectWrap::Unwrap<Domain>(args.This());
+  NanAsyncQueueWorker(new GetMetadataWorker(callback, domain->handle_, type, namespace_uri, flags));
+  NanReturnUndefined();
+}
+
+NLV_WORKER_EXECUTE(Domain, GetMetadata)
+{
+  NLV_WORKER_ASSERT_DOMAIN();
+#ifdef _HAVE_DOMAIN_METADATA_API
```

Comment:
seems to me that the whole worker should be ifdeffed out right? You just bail above if there's no support